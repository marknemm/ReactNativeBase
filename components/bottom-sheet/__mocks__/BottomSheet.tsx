import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
