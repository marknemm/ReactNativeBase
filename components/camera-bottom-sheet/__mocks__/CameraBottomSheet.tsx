import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
