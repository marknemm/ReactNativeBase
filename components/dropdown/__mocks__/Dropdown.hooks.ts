/**
 * Mocks the useDropdownItems hook.
 */
export const useDropdownItems = jest.fn().mockImplementation((data) => data);
