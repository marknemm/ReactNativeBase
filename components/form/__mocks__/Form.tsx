import type { FormProps } from '@components/form/Form.interface';

const FormMock = jest.fn().mockImplementation((props: FormProps) => {
  const ActualComponent = jest.requireActual('@components/form/Form').default;
  return <ActualComponent {...props} />;
});

export default FormMock;
