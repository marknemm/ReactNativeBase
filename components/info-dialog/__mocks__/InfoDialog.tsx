import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
