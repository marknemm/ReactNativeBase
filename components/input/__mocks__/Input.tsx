import type { InputProps } from '@components/input/Input.interfaces';

const MockInput = jest.fn().mockImplementation((props: InputProps) => {
  const ActualComponent = jest.requireActual('@components/input/Input').default;
  return <ActualComponent {...props} />;
});

export default MockInput;
