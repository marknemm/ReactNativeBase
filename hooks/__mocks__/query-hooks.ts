import type { DBFilterOptions, DBFilters, DBQueryOptions, DBQueryOptionsState, DBQueryState, UseQueryOptions } from '@interfaces/db';
import { renderHook } from '@testing-library/react-native';

const ActualModule = jest.requireActual('@hooks/query-hooks');

/**
 * Generates a mock {@link DBQueryOptionsState} from the given {@link DBQueryOptions}.
 *
 * @param options The {@link DBQueryOptions} to mock.
 * @returns The mock {@link DBQueryOptionsState}.
 */
export function genMockQueryOptionsState<
  TData = any,
  TFilters extends DBFilterOptions = DBFilters<TData>,
>(options: DBQueryOptions<TData, TFilters>): DBQueryOptionsState<TData, TFilters> {
  const queryOptionsState = renderHook(() =>
    useQueryOptions(options)
  ).result.current;

  queryOptionsState.setFilters = jest.fn(queryOptionsState.setFilters);
  queryOptionsState.setLimit = jest.fn(queryOptionsState.setLimit);
  queryOptionsState.setOrderBy = jest.fn(queryOptionsState.setOrderBy);
  queryOptionsState.setStartAfter = jest.fn(queryOptionsState.setStartAfter);

  return queryOptionsState as any;
}

/**
 * Mock of custom hook for creating {@link DBQueryOptions} state.
 *
 * @param args The initial {@link DBQueryOptions}.
 * @returns The mock {@link DBQueryOptionsState}.
 */
export const useQueryOptions = jest.fn<DBQueryOptionsState, [DBQueryOptions]>().mockImplementation(
  ActualModule.useQueryOptions
);

type UseQueryParamList = [string, Partial<DBQueryOptionsState>, UseQueryOptions];

/**
 * Mock of query item.
 */
export interface MockQueryItem {

  /**
   * The description of the item.
   */
  description?: string;

  /**
   * The name of the item.
   */
  name: string;

  /**
   * The ID of the item.
   */
  id: number;

  /**
   * Whether the item is public.
   */
  public: boolean;

}

/**
 * Generates a {@link MockQueryItem} list.
 *
 * @param count The number of items to generate. Default is `4`.
 * @returns The {@link MockQueryItem} list.
 */
export function genMockQueryItems(count = 4): MockQueryItem[] {
  const mockQueryItems = [];

  for (let i = 0; i < count; i++) {
    mockQueryItems.push({
      description: `Test description ${i}`,
      id: i,
      name: `Test Item ${i}`,
      public: i % 2 === 0,
    });
  }

  return mockQueryItems;
}

/**
 * Generates a mock {@link DBQueryState} from the given data.
 *
 * @param mockData The mock data to use.
 * @returns The mock {@link DBQueryState}.
 */
export function genMockUseQueryReturn(mockData?: Partial<DBQueryState<MockQueryItem>>): DBQueryState<MockQueryItem> {
  return {
    cancel: jest.fn(),
    cursor: undefined,
    items: genMockQueryItems(),
    loadError: '',
    loading: false,
    loadingInitial: false,
    loadingMore: false,
    loadingOnOptionsChange: false,
    loadNext: jest.fn(),
    refresh: jest.fn(),
    refreshing: false,
    ...mockData,
  };
}

/**
 * Mock of custom hook for querying a list of documents.
 *
 * @param args The {@link UseQueryParamList}.
 * @returns The mock {@link DBQueryState}.
 */
export const useQuery = jest.fn<DBQueryState<MockQueryItem>, UseQueryParamList>().mockReturnValue(
  genMockUseQueryReturn()
);
