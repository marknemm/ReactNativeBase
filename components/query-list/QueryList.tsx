import ActivityIndicator from '@components/activity-indicator/ActivityIndicator';
import ErrorText from '@components/error-text/ErrorText';
import SearchBar from '@components/search-bar/SearchBar';
import { useQuery } from '@hooks/query-hooks';
import { DBFilter } from '@interfaces/db';
import { toTitleCase } from '@util/string';
import { useCallback } from 'react';
import { FlatList } from 'react-native';
import type { QueryListFC, QueryListProps } from './QueryList.interfaces';

/**
 * A list of dynamically queried items.
 *
 * @param props The component {@link QueryListProps}.
 * @returns The {@link QueryList} component.
 */
const QueryList: QueryListFC = ({
  collectionPath,
  debounceMs,
  load,
  map,
  onLoadComplete,
  onLoadError,
  onLoadSuccess,
  queryOptionsState,
  searchFilterName,
  searchFilterOperator = 'starts-with-i',
  searchPlaceholder,
  ...flatListProps
}) => {
  const queryState = useQuery(collectionPath, queryOptionsState, {
    debounceMs,
    load,
    map,
    onLoadComplete,
    onLoadError,
    onLoadSuccess,
  });

  const collectionName = toTitleCase(collectionPath.split('/').pop());
  searchPlaceholder ??= `Search ${collectionName}...`;

  const onSearchChangeCb = useCallback((text: string) => {
    queryOptionsState.setFilters((prevFilters) => ({
      ...prevFilters,
      [searchFilterName]: {
        operator: searchFilterOperator,
        value: text,
      },
    }));
  }, [queryOptionsState, searchFilterName, searchFilterOperator]);

  return (
    <>
      {searchFilterName && (
        <SearchBar
          placeholder={searchPlaceholder}
          onChangeText={onSearchChangeCb}
          showLoading={queryState.loadingOnOptionsChange}
          value={(queryOptionsState.filters[searchFilterName] as DBFilter)?.value ?? ''}
        />
      )}

      <ActivityIndicator
        isVisible={queryState.loadingInitial}
        size="large"
        testID="rnb-query-list-loading-initial"
      />

      <ErrorText
        center
        error={queryState.loadError}
      />

      <FlatList
        data={queryState.items}
        keyExtractor={useCallback((item: any) =>
          item.id,
        [])}
        onEndReached={queryState.loadNext}
        onRefresh={queryState.refresh}
        refreshing={queryState.refreshing}
        {...flatListProps}
      />

      <ActivityIndicator
        isVisible={queryState.loadingMore}
        size="large"
        testID="rnb-query-list-loading-more"
      />
    </>
  );
};

export type * from './QueryList.interfaces';
export default QueryList;
