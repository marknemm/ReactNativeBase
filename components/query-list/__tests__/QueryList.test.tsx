import QueryList from '@components/query-list/QueryList';
import { genMockQueryItems, genMockQueryOptionsState, genMockUseQueryReturn, type MockQueryItem } from '@hooks/__mocks__/query-hooks';
import type { DBQueryOptionsState } from '@interfaces/db';
import { Text } from '@rneui/themed';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen, userEvent } from '@testing-library/react-native';
import { useQuery } from '@hooks/query-hooks';

jest.mock('@hooks/query-hooks');

describe('<QueryList />', () => {
  const collectionPath = 'TestCollection';
  const renderItem = ({ item }: { item: MockQueryItem }) => <Text>{item.name}</Text>;
  const searchPlaceholder = 'Search Test Collection...';
  const searchValue = 'test';
  let queryOptionsState: DBQueryOptionsState<MockQueryItem>;

  beforeEach(() => {
    queryOptionsState = genMockQueryOptionsState({
      limit: 10,
      orderBy: 'name',
    });
  });

  describe('query state and result', () => {
    it('lists all items on successful query', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
        />,
        { wrapper: AppProvider }
      );

      for (const mockItem of genMockQueryItems()) {
        const itemText = screen.getByText(mockItem.name);
        expect(itemText).toBeVisible();
      }
    });

    it('renders error message on failed query', () => {
      const loadError = 'Test error message';
      (useQuery as jest.Mock).mockReturnValueOnce(genMockUseQueryReturn({
        items: [],
        loadError,
      }));

      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
        />,
        { wrapper: AppProvider }
      );

      const errorMessage = screen.getByText(loadError);
      expect(errorMessage).toBeVisible();
    });

    it('renders initial loading spinner while querying initial items', () => {
      (useQuery as jest.Mock).mockReturnValueOnce(genMockUseQueryReturn({
        items: [],
        loading: true,
        loadingInitial: true,
      }));

      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
          searchFilterName="name"
        />,
        { wrapper: AppProvider }
      );

      const loadingInitSpinner = screen.getByTestId('rnb-query-list-loading-initial');
      expect(loadingInitSpinner).toBeVisible();

      const loadingMoreSpinner = screen.queryByTestId('rnb-query-list-loading-more');
      expect(loadingMoreSpinner).toBeFalsy();

      const loadingSearchSpinner = screen.queryByTestId('rnb-search-bar-loading');
      expect(loadingSearchSpinner).toBeFalsy();
    });

    it('renders loading spinner while querying more items', () => {
      (useQuery as jest.Mock).mockReturnValueOnce(genMockUseQueryReturn({
        items: genMockQueryItems(),
        loading: true,
        loadingMore: true,
      }));

      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
          searchFilterName="name"
        />,
        { wrapper: AppProvider }
      );

      const loadingMoreSpinner = screen.getByTestId('rnb-query-list-loading-more');
      expect(loadingMoreSpinner).toBeVisible();

      const loadingInitSpinner = screen.queryByTestId('rnb-query-list-loading-initial');
      expect(loadingInitSpinner).toBeFalsy();

      const loadingSearchSpinner = screen.queryByTestId('rnb-search-bar-loading');
      expect(loadingSearchSpinner).toBeFalsy();
    });

    it('renders loading spinner while searching for items', () => {
      (useQuery as jest.Mock).mockReturnValueOnce(genMockUseQueryReturn({
        items: genMockQueryItems(),
        loading: true,
        loadingOnOptionsChange: true,
      }));

      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
          searchFilterName="name"
        />,
        { wrapper: AppProvider }
      );

      const loadingSearchSpinner = screen.getByTestId('rnb-search-bar-loading');
      expect(loadingSearchSpinner).toBeVisible();

      const loadingInitSpinner = screen.queryByTestId('rnb-query-list-loading-initial');
      expect(loadingInitSpinner).toBeFalsy();

      const loadingMoreSpinner = screen.queryByTestId('rnb-query-list-loading-more');
      expect(loadingMoreSpinner).toBeFalsy();
    });
  });

  describe('search bar', () => {
    it('does not render search bar by default', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          renderItem={renderItem}
          queryOptionsState={queryOptionsState}
        />,
        { wrapper: AppProvider }
      );

      const searchBar = screen.queryByPlaceholderText(/Search/i);
      expect(searchBar).toBeFalsy();
    });

    it('renders search bar', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
          searchFilterName="name"
        />,
        { wrapper: AppProvider }
      );

      const searchBar = screen.getByPlaceholderText('Search Testcollection...');
      expect(searchBar).toBeVisible();
    });

    it('renders search bar with custom placeholder', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
          searchFilterName="name"
          searchPlaceholder={searchPlaceholder}
        />,
        { wrapper: AppProvider }
      );

      const searchBar = screen.getByPlaceholderText(searchPlaceholder);
      expect(searchBar).toBeVisible();
    });

    it('filters items by search input', async () => {
      let newValue;

      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
          searchFilterName="name"
        />,
        { wrapper: AppProvider }
      );

      (queryOptionsState.setFilters as jest.Mock).mockImplementation((cb) => {
        newValue = cb({ description: { operator: 'starts-with-i', value: 'description' } });
        return newValue;
      });

      const searchInput = screen.getByPlaceholderText(/Search/i);
      await userEvent.type(searchInput, searchValue);

      expect(newValue).toEqual({
        description: { operator: 'starts-with-i', value: 'description' },
        name: { operator: 'starts-with-i', value: searchValue },
      });
    });
  });

  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
        />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with search bar', () => {
      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
          searchFilterName="name"
          searchPlaceholder={searchPlaceholder}
        />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly on initial loading', () => {
      (useQuery as jest.Mock).mockReturnValueOnce(genMockUseQueryReturn({
        items: [],
        loading: true,
        loadingInitial: true,
      }));

      render(
        <QueryList
          collectionPath={collectionPath}
          queryOptionsState={queryOptionsState}
          renderItem={renderItem}
        />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
