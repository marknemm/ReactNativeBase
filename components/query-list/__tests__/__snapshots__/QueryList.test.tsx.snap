// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryList /> snapshots renders correctly 1`] = `
<RCTScrollView
  data={
    [
      {
        "description": "Test description 0",
        "id": 0,
        "name": "Test Item 0",
        "public": true,
      },
      {
        "description": "Test description 1",
        "id": 1,
        "name": "Test Item 1",
        "public": false,
      },
      {
        "description": "Test description 2",
        "id": 2,
        "name": "Test Item 2",
        "public": true,
      },
      {
        "description": "Test description 3",
        "id": 3,
        "name": "Test Item 3",
        "public": false,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[MockFunction]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onRefresh={[MockFunction]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[MockFunction]}
      refreshing={false}
    />
  }
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Test Item 0
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Test Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Test Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#242424",
            "fontFamily": undefined,
            "fontSize": 23.57142857142857,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "marginBottom": 0,
            "textAlign": "left",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
          }
        }
      >
        Test Item 3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<QueryList /> snapshots renders correctly on initial loading 1`] = `
[
  <ActivityIndicator
    size="large"
    testID="rnb-query-list-loading-initial"
  />,
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[MockFunction]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>,
]
`;

exports[`<QueryList /> snapshots renders correctly with search bar 1`] = `
[
  <View
    style={
      {
        "marginBottom": 6.285714285714286,
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "height": 98.9730612244898,
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#86939e",
                      "fontSize": 18,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "fontFamily": "material",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="RNE__ICON__Component"
              >
                î¢¶
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        editable={true}
        keyboardAppearance="light"
        onChangeText={[Function]}
        placeholder="Search Test Collection..."
        placeholderTextColor="#86939e"
        style={
          {
            "color": "#242424",
            "flex": 1,
            "fontSize": 23.57142857142857,
            "height": 98.9730612244898,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        {
          "color": "#ff3b30",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>,
  <RCTScrollView
    data={
      [
        {
          "description": "Test description 0",
          "id": 0,
          "name": "Test Item 0",
          "public": true,
        },
        {
          "description": "Test description 1",
          "id": 1,
          "name": "Test Item 1",
          "public": false,
        },
        {
          "description": "Test description 2",
          "id": 2,
          "name": "Test Item 2",
          "public": true,
        },
        {
          "description": "Test description 3",
          "id": 3,
          "name": "Test Item 3",
          "public": false,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[MockFunction]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
              "fontFamily": undefined,
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 0,
              "textAlign": "left",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            }
          }
        >
          Test Item 0
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
              "fontFamily": undefined,
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 0,
              "textAlign": "left",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            }
          }
        >
          Test Item 1
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
              "fontFamily": undefined,
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 0,
              "textAlign": "left",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            }
          }
        >
          Test Item 2
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#242424",
              "fontFamily": undefined,
              "fontSize": 23.57142857142857,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "marginBottom": 0,
              "textAlign": "left",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
            }
          }
        >
          Test Item 3
        </Text>
      </View>
    </View>
  </RCTScrollView>,
]
`;
